modules/gsp-namespace-role/iam.tf

```diff
@@ -0,0 +1,40 @@
+data "aws_iam_policy_document" "assume-role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_id}:role/${var.cluster_name}_kiam_server"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "namespace-sqs" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "sqs:SendMessage",
+      "sqs:ReceiveMessage"
+    ]
+
+    resources = ["arn:aws:sqs:*:${var.account_id}:${var.cluster_name}-sqsqueue-${var.namespace_name}-*"]
+  }
+}
+
+resource "aws_iam_role" "namespace-sqs" {
+  name               = "${var.cluster_name}-namespace-${var.namespace_name}-sqs"
+  assume_role_policy = "${data.aws_iam_policy_document.assume-role.json}"
+  path               = "/gsp/${var.cluster_name}/namespaceroles/sqs/"
```

Comment:
My suggestions for a `path` was instead of doing crazy namespace prefixing, not as well as 😄 

I think `path` is an interesting feature if we're clear what we're going to use it for - and i am the first to admit i was not clear in my original comment.

Reply:
Don't the names still have to be unique?

Reply:
d'oh. yes they do. 😞